tests/test_store.py

```diff
@@ -214,6 +220,35 @@ def test_view_query(
         # tests that the query was valid and successful, that a view subset has actually been retrived
         assert synapse_store_special_scope.storageFileviewTable.empty is False
 
+    @pytest.mark.parametrize(
+        "asset_view,columns,expectation",
+        [
+            ("syn62339865", ["path"], pytest.raises(ValueError)),
+            ("syn62340177", ["id"], pytest.raises(ValueError)),
+            ("syn23643253", ["id", "path"], does_not_raise()),
```

Comment:
Nit: When you're doing tests, you ideally want to split off the testing the "error" vs the non errors.  For example:

- ```def test_view_query_exception_errors()``` <- This would contain all the errors
- ```def test_view_query()``` <- This would be the success.  This would allow you to not have to do `does_not_raise()` function.

Separately, I understand from reading the code and the tests that you're catching for these two conditions

* "Unknown column path"
* "Unknown column"

You may do something like this
```pytest.raises(ValueError, match=r"The path column has not been added to the fileview. Please .*"```

I encourage these tests to be split up, because then youre parametrize, could take `error_type` and `message` as parameters and you could include in the test itself `pytest.raises`

Reply:
Side note, when I review code, the first thing I'm looking at is the tests, because it _should_ tell me exactly what the code's behavior is.

View #1469 for an example of what I personally think are well formatted tests.

Reply:
@thomasyu888 that makes sense and thanks for the linked example. Both were really helpful!